http.go

```diff
@@ -1106,8 +1106,8 @@ func (resp *Response) Reset() {
 	if responseBodyPoolSizeLimit >= 0 && resp.body != nil {
 		resp.ReleaseBody(responseBodyPoolSizeLimit)
 	}
-	resp.Header.Reset()
 	resp.resetSkipHeader()
+	resp.Header.Reset()
```

Comment:
This change is not described in your pull request, why did you change this order?

Reply:
This change is required because the headers need to be reset after the body stream is cleaned up. Otherwise the information whether the connection shall be closed would not be available